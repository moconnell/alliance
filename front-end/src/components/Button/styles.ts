import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background: #ff3864;
  border-radius: 6px;
`;

export const ButtonComponent = styled.button`
  padding: 8px 16px;
  color: #fff;
  font-family: Inter;
  font-size: 14px;
  font-weight: 700;
  line-height: 20px;
  width: 100%;

  :hover {
    opacity: 0.7;
  }

  ${({ disabled }: any) =>
    disabled &&
    `
      pointer-events: none;
      opacity: 0.7;

      :hover{
      opacity: 0.7;
      }
    `}
`;
