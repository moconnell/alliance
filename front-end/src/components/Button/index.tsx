import React from "react";
import { Container, ButtonComponent } from "./styles";

interface ButtonProps {
  children?: React.ReactNode;
  title?: string;
  disabled?: boolean;
  testId?: string;
  type?: "button" | "submit" | "reset";
  onClick?: React.MouseEventHandler;
}

const Button = ({
  children,
  disabled,
  title,  
  testId = "button",
  type = "button",
  onClick,
}: ButtonProps) => {
  return (
    <Container aria-disabled={disabled}>
      <ButtonComponent data-testid={testId} disabled={disabled} title={title} onClick={onClick} type={type}>
        {children}
      </ButtonComponent>
    </Container>
  );
};

export default Button;
