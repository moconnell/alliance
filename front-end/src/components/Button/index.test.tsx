import { render } from "@testing-library/react";
import Button from "./index";

describe("<Button />", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
