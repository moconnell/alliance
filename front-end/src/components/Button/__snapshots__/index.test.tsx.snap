// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB gIoxQ"
  >
    <button
      class="sc-gueYoa bTpkmV"
      data-testid="button"
      type="button"
    />
  </div>
</DocumentFragment>
`;
